Formula/unisonlang.rb

```diff
@@ -0,0 +1,49 @@
+require "language/node"
+
+class Unisonlang < Formula
+  desc "Friendly programming language from the future"
+  homepage "https://unison-lang.org/"
+  url "https://github.com/unisonweb/unison.git",
+      tag:      "release/M3",
+      revision: "97cce8d23147df30b0f80ae745968db09f4e1a44"
+  version "1.0.0-M3"
+  license "MIT"
+  head "https://github.com/unisonweb/unison.git", branch: "trunk"
+
+  depends_on "haskell-stack" => :build
+  depends_on "node" => :build
+
+  on_linux do
+    depends_on "xz" => :build
+  end
+
+  resource "unison-ui" do
+    url "https://github.com/unisonweb/codebase-ui.git",
+        tag:      "release/M3",
+        revision: "05df3a235dec7d4a03ff1c588a32eb6daca288d4"
+  end
+
+  def install
+    # Build the ucm binary
+    system "stack", "build", "--flag", "unison-parser-typechecker:optimized"
+
+    # Install the ucm binary
+    local_install_root = Pathname.new `stack path | awk '/local-install-root/{print $2}'`.to_s.strip
+    cp local_install_root/"bin/unison", prefix/"ucm"
+    bin.install_symlink prefix/"ucm"
+
+    # Build the web interface
+    resource("unison-ui").stage buildpath/"build-ui"
+    cd "build-ui" do
+      system "npm", "install", *Language::Node.local_npm_install_args
+      system "npm", "run", "build"
+    end
+
+    # Install the web interface
+    prefix.install "build-ui/dist/unisonLocal" => "ui"
+  end
+
+  test do
+    assert_match "ucm version:", shell_output("#{bin}/ucm version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.